src/core/Akka/Actor/ActorTelemetry.cs

```diff
@@ -0,0 +1,104 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="ActorTelemetry.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2022 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2022 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using Akka.Event;
+
+namespace Akka.Actor
+{
+    /// <summary>
+    /// A set of events designed to provide some basic telemetry functions for monitoring actor lifecycles.
+    ///
+    /// We want to track actor starts, stops, and restarts. More detailed metrics, such as mailbox size or message
+    /// processing rates will require something like Phobos [https://phobos.petabridge.com/].
+    /// </summary>
+    /// <remarks>
+    /// Not intended to be sent across network boundaries - should only be processed locally via the <see cref="EventStream"/>.
+    /// </remarks>
+    public interface IActorTelemetryEvent : INoSerializationVerificationNeeded, INotInfluenceReceiveTimeout
```

Comment:
These are meant to be metric deltas / increments that can be aggregated by a subscriber on the `EventStream` - not used for tracing or logs. Specific to the `IActorRef` lifecycle.

Reply:
Handling matters like measuring message processing rates this way would be a disaster from a performance and memory management point of view - the overhead of measuring message processing rates would be greater than or equal to the overhead of processing the messages themselves. Use [Phobos](https://phobos.petabridge.com/) for that use case.